### PR TITLE
CI: align test flags in various config files

### PR DESCRIPTION
**Pull request by mback2k**
1. Remove redundant parameters -a -p from target test-nonflaky.
2. Disable testing of SCP protocol on native Windows environments.
3. Use Makefile target to run tests in autotools builds on AppVeyor.
4. Don't use -vc parameter which is reserved for debugging.

Replaces #7591

--

The other things mentioned in #7591 will be done separately.

### CODE REVIEW

**Comment by mback2k** [comment]
The zuul CI failure is unrelated to this PR as those CI files aren't touched.

File: .azure-pipelines.yml [hunk 1]
```diff
@@ -203,4 +203,4 @@ stages:
       displayName: 'test'
       env:
         AZURE_ACCESS_TOKEN: "$(System.AccessToken)"
-        TFLAGS: "-u -vc /usr/bin/curl.exe -r -rm $(tests)"
+        TFLAGS: "-r -rm -u !SCP $(tests)"
```
Comment (bagder) [inline .azure-pipelines.yml:206]: Does anyone ever read the `-r` generated output? I know I never do it.

**Review by bagder — COMMENTED** [review]
I'm not sure I see the benefit of this, but if you think it makes things better go for it.

I think maybe added conditional check for appveyor is a bit hidden here while probably being the larger actual change!

File: .azure-pipelines.yml (hunk 1 above)
Comment (mback2k) [reply .azure-pipelines.yml:206]: I do sometimes as one way to figure out potential reasons for test flakiness and timeouts.

**Pull request closed without merging**